src/action.js

```diff
@@ -25,10 +31,16 @@ async function action(context = github.context) {
       )
     }
 
-    const ref = context.payload.pull_request.head.ref
-    const config = {
-      ...defaults,
-      ...(await getConfig(octokit, CONFIG_FILENAME, repoInfo, ref))
+    const ref = context.payload.pull_request.head.ref;
+
+    /**
+     * load custom config when existing or
+     * set default config when no custom overwrite exists
+     */
+    var config = defaults;
+    var customConfig = await getConfig(octokit, customConfigFile, repoInfo, ref);
+    if(customConfig !== null) {
+        config = customConfig;
```

Comment:
Can we move this logic to `getConfig`? We can add an extra parameter for the default config which it will only return if `error.code === 404` :)

```js
  const config = await getConfig(octokit, customConfigFile, repoInfo, ref, defaultConfig);
```